internal/support/scheduler/init.go

```diff
@@ -0,0 +1,265 @@
+	conf := &ConfigurationStruct{}
+	err := config.LoadFromFileV2(useProfile, conf)
+	if err != nil {
+		return nil, err
+	}
+
+	if useConsul {
+		conf, err = connectToConsul(conf)
+		if err != nil {
+			return nil, err
+		}
+	}
+	return conf, nil
+}
+
+func connectToConsul(conf *ConfigurationStruct) (*ConfigurationStruct, error) {
+	//Obtain ConsulConfig
+	cfg := consulclient.NewConsulConfig(conf.Registry, conf.Service, internal.SupportSchedulerServiceKey)
+
+	// Register the service in Consul
+	err := consulclient.ConsulInit(cfg)
+
+	if err != nil {
+		return conf, fmt.Errorf("connection to Consul could not be made: %v", err.Error())
+	}
+	// Update configuration data from Consul
+	updateCh := make(chan interface{})
+	errCh := make(chan error)
+	dec := consulclient.NewConsulDecoder(conf.Registry)
+	dec.Target = &ConfigurationStruct{}
+	dec.Prefix = internal.ConfigV2Stem + internal.SupportSchedulerServiceKey
+	dec.ErrCh = errCh
+	dec.UpdateCh = updateCh
+
+	defer dec.Close()
+	defer close(updateCh)
+	defer close(errCh)
+	go dec.Run()
+
+	select {
+	case <-time.After(2 * time.Second):
+		err = errors.New("timeout loading config from registry")
+	case ex := <-errCh:
+		err = errors.New(ex.Error())
+	case raw := <-updateCh:
+		actual, ok := raw.(*ConfigurationStruct)
+		if !ok {
+			return conf, errors.New("type check failed")
+		}
+		conf = actual
+	}
+
+	return conf, err
+}
+
+func listenForConfigChanges() {
+	errCh := make(chan error)
+	dec := consulclient.NewConsulDecoder(Configuration.Registry)
+	dec.Target = &ConfigurationStruct{}
+	dec.Prefix = internal.ConfigV2Stem + internal.SupportSchedulerServiceKey
+	dec.ErrCh = errCh
+	dec.UpdateCh = chConfig
+
+	defer dec.Close()
+	defer close(errCh)
+
+	go dec.Run()
+	for {
+		select {
+		case ex := <-errCh:
+			LoggingClient.Error(ex.Error())
+		case raw, ok := <-chConfig:
+			if ok {
+				actual, ok := raw.(*ConfigurationStruct)
+				if !ok {
+					LoggingClient.Error("listenForConfigChanges() type check failed")
+				}
+				Configuration = actual //Mutex needed?
+			} else {
+				return
+			}
+		}
+	}
+}
+
+func initializeClients(useConsul bool) {
+	// Create metadata clients
+	params := types.EndpointParams{
+		ServiceKey:  internal.SupportSchedulerServiceKey,
+		Path:        clients.ApiDeviceRoute,
+		UseRegistry: useConsul,
+		Url:         Configuration.Clients["Metadata"].Url() + clients.ApiDeviceRoute,
+		Interval:    Configuration.Service.ClientMonitor,
+	}
+
+	mdc = metadata.NewDeviceClient(params, startup.Endpoint{})
+	params.Path = clients.ApiCommandRoute
+	params.Url = Configuration.Clients["Metadata"].Url() + clients.ApiCommandRoute
+	cc = metadata.NewCommandClient(params, startup.Endpoint{})
+}
+
+func setLoggingTarget() string {
+	if Configuration.Logging.EnableRemote {
+		return Configuration.Clients["Logging"].Url() + clients.ApiLoggingRoute
+	}
+	return Configuration.Logging.File
+}
+
+func AddDefaultSchedulers() error {
+
```

Comment:
Consider moving this function to schedule.go. Add unit tests.

Reply:
Moved this code into the Scheduler.  Were no longer using a bridge from main to inti too scheduler.